skyportal/tests/flaky/test_frontend_flaky.py

```diff
@@ -355,3 +355,108 @@ def test_gcnevents_observations(
     driver.wait_for_xpath('//*[text()="ztfr"]')
     driver.wait_for_xpath('//*[text()="1.57415"]')
     driver.wait_for_xpath('//*[text()="20.40705"]')
+
+
+@pytest.mark.flaky(reruns=2)
+def test_gcn_request(driver, user, super_admin_token, public_group):
+
+    datafile = f'{os.path.dirname(__file__)}/../data/GW190425_initial.xml'
+    with open(datafile, 'rb') as fid:
+        payload = fid.read()
+    data = {'xml': payload}
+
+    status, data = api('POST', 'gcn_event', data=data, token=super_admin_token)
+    assert status == 200
+    assert data['status'] == 'success'
+
+    telescope_name = str(uuid.uuid4())
+    status, data = api(
+        'POST',
+        'telescope',
+        data={
+            'name': telescope_name,
+            'nickname': telescope_name,
+            'lat': 0.0,
+            'lon': 0.0,
+            'elevation': 0.0,
+            'diameter': 10.0,
+        },
+        token=super_admin_token,
+    )
+    assert status == 200
+    assert data['status'] == 'success'
+    telescope_id = data['data']['id']
+
+    fielddatafile = f'{os.path.dirname(__file__)}/../../../data/ZTF_Fields.csv'
+    regionsdatafile = f'{os.path.dirname(__file__)}/../../../data/ZTF_Region.reg'
+
+    instrument_name = str(uuid.uuid4())
+    status, data = api(
+        'POST',
+        'instrument',
+        data={
+            'name': instrument_name,
+            'type': 'imager',
+            'band': 'NIR',
+            'filters': ['ztfr'],
+            'telescope_id': telescope_id,
+            "api_classname_obsplan": "ZTFMMAAPI",
+            'field_data': pd.read_csv(fielddatafile)[:5].to_dict(orient='list'),
+            'field_region': Regions.read(regionsdatafile).serialize(format='ds9'),
+        },
+        token=super_admin_token,
+    )
+    assert status == 200
+    assert data['status'] == 'success'
+    instrument_id = data['data']['id']
+
+    # wait for the fields to populate
+    time.sleep(15)
+
+    datafile = f'{os.path.dirname(__file__)}/../../../data/sample_observation_data.csv'
+    data = {
+        'telescopeName': telescope_name,
+        'instrumentName': instrument_name,
+        'observationData': pd.read_csv(datafile).to_dict(orient='list'),
+    }
+
+    status, data = api('POST', 'observation', data=data, token=super_admin_token)
+
+    assert status == 200
+    assert data['status'] == 'success'
+
+    # wait for the executed observations to populate
+    time.sleep(15)
```

Comment:
We can definitely do better without this `sleep`. First let the driver get `become_user` and the event page, and after that there's a `wait_for_xpath` anyway.

Reply:
got this one too.